js/src/Map.ts

```diff
@@ -321,133 +353,90 @@ export class LeafletMapView extends utils.LeafletDOMWidgetView {
+    this.listenTo(this.model, 'change:panes', this.rerender);
+    this.listenTo(this.model, 'change:dragging', () => {
+      if (this.model.get('dragging')) {
+        this.obj.dragging.enable();
+      } else {
+        this.obj.dragging.disable();
+      }
+    });
+    this.listenTo(this.model, 'change:layers', () => {
+      this.layer_views.update(this.model.get('layers'));
+    });
+    this.listenTo(this.model, 'change:controls', () => {
+      this.control_views.update(this.model.get('controls'));
+    });
+    this.listenTo(this.model, 'change:zoom', () => {
+      if (!this.dirty) {
+        this.dirty = true;
+        // Using flyTo instead of setZoom to adjust for potential
+        // sub-pixel error in leaflet object's center.
+        //
+        // Disabling animation on updates from the model because
+        // animation triggers a `moveend` event in an animationFrame,
+        // which causes the center to bounce despite of the dirty flag
+        // which is set back to false synchronously.
+        this.obj.flyTo(this.model.get('center'), this.model.get('zoom'), {
+          animate: false,
         });
-      },
-      this
-    );
-    this.listenTo(
-      this.model,
-      'change:dragging_style',
-      function () {
-        this.model.update_style();
-      },
-      this
-    );
-    this.listenTo(
-      this.model,
-      'change:default_style',
-      function () {
-        this.model.update_style();
-      },
-      this
-    );
-    this.listenTo(
-      this.model,
-      'change:fullscreen',
-      function () {
-        var fullscreen = this.model.get('fullscreen');
-        if (this.obj.isFullscreen() !== fullscreen) {
-          this.obj.toggleFullscreen();
-        }
-      },
-      this
-    );
+        this.dirty = false;
+      }
+      this.model.update_bounds().then(() => {
+        this.touch();
+      });
+    });
+    this.listenTo(this.model, 'change:center', () => {
+      if (!this.dirty) {
+        this.dirty = true;
+        this.obj.panTo(this.model.get('center'));
+        this.dirty = false;
+      }
+      this.model.update_bounds().then(() => {
+        this.touch();
+      });
+    });
+    this.listenTo(this.model, 'change:dragging_style', () => {
+      this.model.update_style();
+    });
+    this.listenTo(this.model, 'change:default_style', () => {
+      this.model.update_style();
+    });
+    this.listenTo(this.model, 'change:fullscreen', () => {
+      var fullscreen = this.model.get('fullscreen');
+      if (this.obj.isFullscreen() !== fullscreen) {
+        this.obj.toggleFullscreen();
+      }
+    });
   }
 
-  handle_msg(content) {
-    switch (content.method) {
-      case 'foo':
-        break;
-    }
-  }
+  // handle_msg(content) {
+  //   switch (content.method) {
+  //     case 'foo':
+  //       break;
+  //   }
+  // }
 
-  processPhosphorMessage(msg) {
+  processPhosphorMessage(msg: Message) {
+    //TODO: fix this?
+    //@ts-ignore
```

Comment:
This actually cannot compile (it uses APIs from the archived [Phosphor](https://github.com/phosphorjs/phosphor) library which is now [Lumino](https://github.com/jupyterlab/lumino)). It's used for JupyterLab 3 support.

So we should keep the ts-ignore and document it.

```suggestion
    // @ts-ignore This is for backward compatibility with JupyterLab 3
```